### ISSUE TITLE
Make kubeconfig ttl global settings configurable

### ISSUE THREAD

**Issue by bashofmann** [rancher/dashboard#3044]
With Rancher 2.4.6, two new global settings were added with which you can configure a TTL for kubeconfig tokens:

* kubeconfig-token-ttl-minutes
* kubeconfig-generate-token

Currently, you have to edit them through the API (https://rancher.com/docs/rancher/v2.x/en/api/api-tokens/). It would be great, if they could be exposed in the dashboard as well.

**Comment by gaktive** [comment]
PR: https://github.com/rancher/dashboard/pull/3045

**Comment by izaac** [comment]
Rancher server: `2.6 master-head (07/01/2021)` __
I am getting a `500` when I set the `kubeconfig-generate-token` to `false`
Then clicking download Kubeconfig from a cluster (RKE1 DO)

```
POST /v3/clusters/c-vhrzx?action=generateKubeconfig HTTP/1.1
Host: <RANCHER_SERVER>
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:78.0) Gecko/20100101 Firefox/78.0 Waterfox/78.12.0
Accept: application/json
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate, br
Referer: https://<RANCHER_SERVER>/dashboard/c/local/manager/provisioning.cattle.io.cluster
x-api-csrf: aac9ba5471
Origin: https://<RANCHER_SERVER>
Connection: keep-alive
Cookie: R_PCS=dark; R_LOCALE=en-us; CSRF=aac9ba5471; R_SESS=token-qbfc6:95v9c78lxlhx78rctn26cfsp9m4qmnslmvhftpkxpmfnvjbdf5sdg6
Pragma: no-cache
Cache-Control: no-cache
Content-Length: 0
```

Response:
`{"baseType":"error","code":"ServerError","message":"resource name may not be empty","status":500,"type":"error"}`

**Comment by bashofmann** [comment]
@izaac I'd say this issue is independent from this ticket. This ticket is only about adding the UI to set this setting. The setting itself has been around already in 2.5. Previously you had to set it through the Rancher API (see docs https://rancher.com/docs/rancher/v2.x/en/api/api-tokens/).

If the setting itself is not working anymore in 2.6, I think this is a separate problem, probably even in rancher/rancher and not in rancher/dashboard.

Should this be tracked in this ticket as well, or is it better to create a separate ticket.

**Comment by izaac** [comment]
@bashofmann I will do a comparative test from Ember on master and check 2.5.8 too. I thought something was a miss in the request from dashboard after I checked the response message.

**Comment by bashofmann** [comment]
I just did a quick test today:

```
docker run -d --restart=unless-stopped -p 80:80 -p 443:443 --privileged rancher/rancher:master-head
```

In Rancher I set Global Settings -> `kubeconfig-generate-token`  to `false`.

Then I went to the local cluster and could download a kubeconfig which correctly contained

```
users:
- name: "local"
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1beta1
      args:
        - token
        - --server=localhost
        - --user=local
      command: rancher
```

instead of a user token.

**Comment by izaac** [comment]
@bashofmann yes this seems to only fail on downstream clusters I am opening a separate issue on `rancher/rancher`

**Comment by izaac** [comment]
Some validation is blocked by https://github.com/rancher/rancher/issues/33474

**Comment by izaac** [comment]
Rancher version: `2.6-head (08/04/2021)` _10588ab_
rancher cli version: `v2.4.12-rc2`

I was able to set the `kubeconfig-token-ttl-minutes` and test it with the rancher cli to check the tokens had the configured expiration time.
Validation of `kubeconfig-generate-token` was done as part of: https://github.com/rancher/rancher/issues/33474

Steps for `kubeconfig-token-ttl-minutes` validation:

- Configure a Rancher server to use rancher cli
- On the Rancher Global settings set the `kubeconfig-token-ttl-minutes` value other than default.
- Configure the Rancher CLI to generate the `cli2.json`
- Create a Standard Rancher user, get the user-id
- `rancher token --server <rancher_url> --user user-ljwf6`
- check the json output or login as the user and check the kubeconfig token generated and the expiration time. 
- The TTL Global setting value should match the expiration time displayed for the token

Leaving the setting with the default value, the rancher cli generates the token properly too. (16 Hours)
